### PR TITLE
Don't use Elasticsearch to look up works in the items API

### PR DESCRIPTION
**Pull request by alexwlchan**
For https://github.com/wellcomecollection/platform/issues/5481

This decouples the items API from Elasticsearch. Previously it went:

```mermaid
graph LR
    I[items API] --> ES[Elasticsearch]
```

Now it goes:

```mermaid
graph LR
    I[items API] --> C[catalogue API]
    C --> ES[Elasticsearch]
```

This means changes to the Elasticsearch infrastructure don't affect the items API. The Scala types aren't quite ideal for this, but they're good enough, and at some point we might revisit this more substantially as part of Catalogue API V3. We could do a more extensive refactor here to entirely decouple it from the Catalogue API, but I don't want to bite off too much in one go.

The diff isn't as big as it appears at first blush – a lot of it is de-denting tests where we no longer need to set up as much context.

### CODE REVIEW

**Comment by alexwlchan** [comment]
Oof, building the items API took ~3m in a previous build (https://buildkite.com/wellcomecollection/catalogue-api/builds/1533), now it's done to ~2m. A whole minute just to pull an Elasticsearch container! ⚡

File: common/stacks/src/main/scala/weco/api/stacks/models/SierraItemIdentifier.scala [hunk 1]
```diff
@@ -28,7 +29,7 @@ object SierraItemIdentifier {
       ontologyType = "Item"
     )
 
-  def fromSourceIdentifier(
+  def fromOldSourceIdentifier(
```
Comment (jamieparkinson) [inline common/stacks/src/main/scala/weco/api/stacks/models/SierraItemIdentifier.scala:32]: what does `old` mean?

File: items/src/main/scala/weco/api/items/services/WorkLookup.scala [hunk 2]
```diff
@@ -1,30 +1,68 @@
...
+        case StatusCodes.NotFound =>
+          info(s"Not Found for GET to $path with $params")
+          Future(Left(WorkNotFoundError(id)))
+
+        case StatusCodes.Gone =>
+          info(s"Gone for GET to $path with $params")
+          Future(Left(WorkGoneError(id)))
+
+        case status =>
+          val err = new Throwable(s"$status from bag tracker API")
```
Comment (jamieparkinson) [inline items/src/main/scala/weco/api/items/services/WorkLookup.scala:57]: assuming that `bag tracker` is a bad copy/paste from the storage service?

**Review by jamieparkinson — APPROVED** [review]
Remarkably straightforward to do this - nice

File: common/stacks/src/main/scala/weco/api/stacks/models/SierraItemIdentifier.scala (hunk 1 above)
Comment (alexwlchan) [reply common/stacks/src/main/scala/weco/api/stacks/models/SierraItemIdentifier.scala:32]: "old" is just "the old way of doing things", a change I made to keep the requests API compiling – I didn't want to change too much in one go. The next PR will remove this method when I flip the requests API to use the new types.

File: items/src/main/scala/weco/api/items/services/WorkLookup.scala (hunk 2 above)
Comment (alexwlchan) [reply items/src/main/scala/weco/api/items/services/WorkLookup.scala:57]: Oops, yes. Good spot, thanks!

**Pull request merged**